hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1013,11 +1015,15 @@ Y(lslxis, X, xzr, 64)
 #undef Y
 
 void Vgen::emit(const popp& i) {
-  a->Ldp(X(i.d0), X(i.d1), MemOperand(sp, 16, PostIndex));
+  Register r0 = i.d0.isValid() ? X(i.d0) : vixl::xzr;
+  Register r1 = i.d1.isValid() ? X(i.d1) : vixl::xzr;
+  a->Ldp(r0, r1, MemOperand(sp, 16, PostIndex));
```

Comment:
Here's the implementation of `popp` for ARM. We still use a `ldp` instruction, but we'll pass the special zero register (`$xzf`) for `InvalidReg`.

For X64 and PPC64, I planned to lower `popp` to a `pop` and `lea` of `$sp` when `InvalidReg` is used.